rados/write_element.go

```diff
@@ -0,0 +1,33 @@
+package rados
+
+// #include <stdint.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+type writeElement struct {
+	// inputs:
+	b []byte
+
+	// arguments:
+	cBuffer   *C.char
+	cDataLen  C.size_t
+	cWriteLen C.size_t
+	cOffset   C.uint64_t
+}
+
+func newWriteElement(b []byte, writeLen, offset uint64) *writeElement {
+	return &writeElement{
+		b:         b,
+		cBuffer:   (*C.char)(unsafe.Pointer(&b[0])),
+		cDataLen:  C.size_t(len(b)),
+		cWriteLen: C.size_t(writeLen),
+		cOffset:   C.uint64_t(offset),
+	}
+}
+
+func (*writeElement) reset()        {}
```

Comment:
Maybe add a comment why these can be empty. If there are more occasions for this, you could create an `emptyElement` type and implement these functions there and inherit them.

Reply:
I can do the latter preemptively for clarity.

Reply:
done